portal-impl/src/com/liferay/portal/verify/VerifyProperties.java

```diff
@@ -178,15 +190,18 @@ protected void verifyObsoleteSystemProperty(String key) throws Exception {
 		}
 	}
 
-	protected void verifyPortalProperties() throws Exception {
+	protected List<String> verifyPortalProperties() throws Exception {
+		LinkedList<String> unmigratedKeys = new LinkedList<>();
```

Comment:
@W2jx Without specific reason, we usually declare a variable to be instance of an interface instead of an implementation:

```
List<String> xxx = new XXXList<>();
```

Also, what's the reason to use `LinkedList` here?

Reply:
Ok,I will fix it. The reason is that I want to keep the original error log, So I just need to keep the properties,  either StringBundler or List will do.  And The reason why I chose LinkedList is that I just need to insert and read sequentially, and I don't know the number of properties in advance.

Reply:
@dantewang Can I continue to use LinkedList or replace it with something else?

Reply:
@W2jx That's OK to me. But we may need to double check the overall design later.